Code/Framework/AzCore/Platform/iOS/AzCore/NativeUI/NativeUISystemComponent_iOS.mm

```diff
@@ -58,14 +58,18 @@ static void OnDeadlock(NSString* nsTitle, NSString* nsMessage)
             freopen([pathForLog cStringUsingEncoding:NSASCIIStringEncoding], "a+", stderr);
             
             // These messages go to the file.
-            NSLog(@"BlockingDialog failed to display at %@", dateString);
+            NSLog(@"BlockingDialog might be failed to display at %@", dateString);
```

Comment:
Better message might be something like:  "Main thread is blocked.  This may arise due to the main thread waiting on a semaphore locked by this thread."

Or something like that.  In other words, in the log explain why we're dumping the assert contents (and which situations might trigger it) so that other devs know it's an expected situation.

Reply:
Addressed.